p2p/protocol/head/head.go

```diff
@@ -69,7 +69,9 @@ func QueryRootCid(ctx context.Context, host host.Host, topic string, peerID peer
 					// If protocol ID is wrong, then try the old "double-slashed" protocol ID.
 					//
 					// TODO: remove this code when all providers have upgraded.
-					if !errors.Is(err, multistream.ErrNotSupported) {
+					if !errors.Is(err, multistream.ErrNotSupported[protocol.ID]{
+						Protos: []protocol.ID{deriveProtocolID(topic)},
+					}) {
```

Comment:
I would recommend moving to https://github.com/ipni/go-libipni/dagsync which is the replacement go-legs. If that is not possible, then change this PR to do as the code there does:
https://github.com/ipni/go-libipni/blob/main/dagsync/p2p/protocol/head/head.go#L80-L95